src/gui/options_imp.cpp

```diff
@@ -583,6 +587,13 @@ int options_imp::getProxyType() const
     }
 }
 
+int options_imp::limitStep(int value)
+{
+    static const int pow10[] = { 1, 1, 10, 100, 1000, 10000, 10000 };
+    int log = static_cast<int>(log10(value));
+    return pow10[log];
```

Comment:
Added range check in `adaptLimitStep` by comparing against the max and min values from the spinBox, and throwing out_of_range exception if the value is outside of that.